packages/flutter/lib/src/cupertino/switch.dart

```diff
@@ -173,7 +183,7 @@ class _CupertinoSwitchRenderObjectWidget extends LeafRenderObjectWidget {
     return _RenderCupertinoSwitch(
       value: value,
       activeColor: activeColor,
-      trackColor: CupertinoDynamicColor.resolve(CupertinoColors.secondarySystemFill, context),
+      trackColor: trackColor,
```

Comment:
in `void updateRenderObject(BuildContext context, _RenderCupertinoSwitch renderObject)` (line 195) the track color needs to be updated accordingly.